### PR TITLE
Issue 3895 - SSO demo

### LINKED ISSUE #3895: Create a simple SSO demo page
### Description
Create a simple static html page to demonstrate how the backend API works for SSO.

### Goals
- [x] Front end developers will be able to look at the example page to understand how to implement the logic to interact with the SSO apis

### Tasks
- [x] Create sample journeys for:
  - [x] login
  - [x] sign up
  - [x] link
  - [x] unlink
- [x] Change redirect to just returning a link to microsoft to avoid CORS problem

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #3895

#### Description
- created a SSO demo page to demonstrate the whole workflow (accessible via `{domain}/sso.html`)
- changed all the 302 to microsoft to be 200 with the link to microsoft to avoid CORs (front end will just have to redirect the window as necessary - demonstrated in demo page)

#### Test cases
- all tests should pass
- demo should function as intended

### CODE REVIEW

File: backend/tests/v5/unit/middleware/sso/aad.test.js [hunk 1]
```diff
@@ -261,7 +263,9 @@ const testAuthenticate = () => {
 
 			await Aad.authenticate(redirectUri)(req, res);
 			expect(Sso.addPkceProtection).toHaveBeenCalledTimes(1);
-			expect(Responder.respond).not.toHaveBeenCalled();
+			expect(Responder.respond).toHaveBeenCalledTimes(1);
+			expect(Responder.respond).toHaveBeenCalledWith(req, res, templates.ok, expect.anything());
```
Comment (ChristosTsiotsias) [inline backend/tests/v5/unit/middleware/sso/aad.test.js:267]: could we maybe use

https://jestjs.io/docs/expect#expectobjectcontainingobject

File: backend/tests/v5/e2e/routes/sso/aad/aad.test.js [hunk 2]
```diff
@@ -189,9 +189,9 @@ const signup = () => {
 
 			const res = await agent.post(`/v5/sso/aad/signup?redirectUri=${redirectUri}`)
 				.send(newUserData)
-				.expect(302);
+				.expect(templates.ok.status);
 
-			const resUri = new URL(res.headers.location);
+			const resUri = new URL(res.body.link);
```
Comment (ChristosTsiotsias) [inline backend/tests/v5/e2e/routes/sso/aad/aad.test.js:194]: test description needs update

File: backend/tests/v5/e2e/routes/sso/aad/aad.test.js [hunk 3]
```diff
@@ -292,8 +292,8 @@ const testLink = () => {
 
 			test('should redirect the user to Microsoft authentication page', async () => {
 				const res = await testSession.get(`/v5/sso/aad/link?redirectUri=${redirectUri}`)
-					.expect(302);
-				const resUri = new URL(res.headers.location);
+					.expect(templates.ok.status);
```
Comment (ChristosTsiotsias) [inline backend/tests/v5/e2e/routes/sso/aad/aad.test.js:295]: test description needs update

**Review by ChristosTsiotsias — CHANGES_REQUESTED** [review]
I think we still need to update the swagger documentation of the affected endpoints

File: backend/tests/v5/unit/middleware/sso/aad.test.js (hunk 1 above)
Comment (carmenfan) [reply backend/tests/v5/unit/middleware/sso/aad.test.js:267]: thought we couldn't but i guess `expect.any()` inside the link property would work here! 👍

**Review by ChristosTsiotsias — APPROVED** [review]

**Pull request merged**